build.sh

```diff
@@ -1,96 +1,345 @@
 #!/usr/bin/env bash
 
+usage()
+{
+    echo "Usage: $0 [managed] [native] [BuildArch] [BuildType] [clean] [verbose] [clangx.y]"
+    echo "manged - optional argument to build the managed code"
+    echo "native - optional argument to build the native code"
+    echo "The following arguments affect native builds only:"
```

Comment:
Correct, but I tried to keep the delta small with this PR. We can always extend the functionality from this point on.